server/master/src/main/java/org/apache/accumulo/master/tableOps/tableImport/MoveExportedFiles.java

```diff
@@ -47,25 +47,27 @@
     try {
       VolumeManager fs = master.getFileSystem();
 
-      Map<String,String> fileNameMappings = PopulateMetadataTable.readMappingFile(fs, tableInfo);
+      for (ImportedTableInfo.DirectoryMapping dm : tableInfo.directories) {
+        Map<String,String> fileNameMappings = PopulateMetadataTable.readMappingFile(fs,
+            dm.importDir);
 
-      for (String oldFileName : fileNameMappings.keySet()) {
-        if (!fs.exists(new Path(tableInfo.exportDir, oldFileName))) {
-          throw new AcceptableThriftTableOperationException(tableInfo.tableId.canonical(),
-              tableInfo.tableName, TableOperation.IMPORT, TableOperationExceptionType.OTHER,
-              "File referenced by exported table does not exists " + oldFileName);
+        for (String oldFileName : fileNameMappings.keySet()) {
+          if (!fs.exists(new Path(dm.exportDir, oldFileName))) {
```

Comment:
This seems like an issue with the existing code and could be fixed in another PR.  This exists call seems unnecessary if the call to rename would fail when the file did not exists.  Seems like less namenode RPCs could be made.  Or maybe the array files below could be compared to the key set of the map.

Reply:
Good idea, I'll take a pass a this.

Reply:
If you do not get to it I will open up an issue.